src/main/java/com/faforever/client/remote/FafServerAccessorImpl.java

```diff
@@ -302,13 +303,16 @@ protected void cancelled() {
   }
 
   @Override
-  @PreDestroy
-  public void disconnect() {
+  public void destroy() {
     if (fafConnectionTask != null) {
       fafConnectionTask.cancel(true);
     }
   }
 
+  public void disconnect(){
```

Comment:
Same here, make `destroy()` call `disconnect()` and move the code over. Otherwise, you're breaking things.
Also, there's a missing space between `){`. Please enable automatic code formatting in your commit dialog and fix the 3 occurrences in this PR.